### DISCUSSION TITLE
Plans for iOS supports

### DISCUSSION THREAD

**Discussion by norvelx** [limbonaut/limboai#232] [General]
I am currently developing a small game targeting the iOS platform. I noticed a pull request related to iOS support ([PR #163](https://github.com/limbonaut/limboai/pull/163)) and was wondering whether I should start building my NPCs using LimboAI, expecting that iOS support will be added soon, or if it would be better to use other behavior tree add-ons for now.

I also have a question about how difficult it might be to compile the game for platforms like PlayStation or Nintendo Switch, if that becomes necessary. I understand that porting to platforms like PlayStation is usually done by third-party companies. In that case, could using LimboAI create any additional challenges for those handling the port?

Thank you for your time, and I look forward to your response.

**Comment by limbonaut** [comment]
For iOS, it's a matter of making the GitHub actions produce builds that work on export. We already have export templates for iOS in the LimboAI module variant - those should work as expected. But GDExtension is not yet complete, as some folks reported that iOS builds from that PR don't work on export, and since I don't have an iPhone - I can't figure out why it breaks. I had some ideas I wanted to try as soon as I get some time for it, but I need someone to test it. If you are experienced with compiling, and have macOS device, you can try compiling builds on your own from existing sources. You'll need Xcode, SCons tool and that's all I think.

For console builds, I don't have access to those, but I believe that dropping sources into `godot/modules/limboai` and compiling a console port as usual would work with 98% confidence.

**Reply by norvelx** [reply]
Hello. Thank you for feedback. I have MacBook and iPhone but I have zero knowledge about compiling stuff in Xcode. I can try to export game to iOS and checks compile errors. Maybe that will give me some clues.

**Reply by limbonaut** [reply]
Currently, the iOS libraries are not included in GDExtension builds because the exported project didn't work in testing. The ios libs can be compiled manually with `scons target=template_release platform=ios`.

**Reply by norvelx** [reply]
Hello. I compiled libs for iOS. I can export a game to the iPhone and it works but I see an errors in Xcode.


```
ERROR: Error loading GDExtension configuration file: res://addons/limboai/TMPbin/limboai.gdextension
at: load_gdextension_resource (core/extension/gdextension.cpp:927)
ERROR: Failed loading resource: res://addons/limboai/TMPbin/limboai.gdextension. Make sure resources have been imported by opening the project in the editor at least once.
at: _load (core/io/resource_loader.cpp:284)
ERROR: Error loading extension: res://addons/limboai/TMPbin/limboai.gdextension
at: load_extensions (core/extension/gdextension_manager.cpp:234)
```

Software version I used:

Xcode 16.0
Godot 4.3
iOS 18.0.1
MacOS 15.0.1
SCons: v4.8.1.08661ed4c552323ef3a7f0ff1af38868cbabb05e,

I compiled like this..

```
scons arch=arm64 target=template_release platform=ios
scons arch=arm64 target=template_debug platform=ios
scons arch=universal target=template_debug platform=ios
scons arch=universal target=template_release platform=iOS
```
Copied files to bin folder.
Added to limboai.gdextension

ios.release.arm64 = "res://addons/limboai/bin/liblimboai.ios.template_release.arm64.dylib"
ios.debug.arm64 = "res://addons/limboai/bin/liblimboai.ios.template_debug.arm64.dylib"
ios.release.simulator = "res://addons/limboai/bin/liblimboai.ios.template_release.universal.dylib"
ios.debug.simulator = "res://addons/limboai/bin/liblimboai.ios.template_debug.universal.dylib"


Also I see that [issue godotengine/godot/issues/90168](https://github.com/godotengine/godot/issues/90168) is closed now.
Maybe mentioned issue is not relevant in final release of Godot 4.3.

**Reply by limbonaut** [reply]
Maybe... Hmm. I can dispatch a build in the PR. Would you like to test it?

**Reply by limbonaut** [reply]
I rebased and updated the PR: https://github.com/limbonaut/limboai/pull/163
Test Build: https://github.com/limbonaut/limboai/actions/runs/11365734230

**Reply by norvelx** [reply]
I’ll test it asap.

**Reply by norvelx** [reply]
I tried new Test Build: [https://github.com/limbonaut/limboai/actions/runs/11365734230](https://github.com/limbonaut/limboai/actions/runs/11365734230)

I Exported project to IOS with no errors.
Also when I launch project through Xcode I see no errors as well. Errors that I mentioned in previous comments is also gone.
Deploying project to iOS device from Godot editor directly also works as expected.

From my side everything looks smooth.
If you have some test cases you want me to try please let me know.

Something worse to mention which is not related to this specific build.
When you install LimboAI from Godot Assets tab (inside Godot editor) there are no problems related to MacOS security feature.(protection from "unidentified developer")

If you try to copy manually limboai folder to addons folder of a Godot project you will get this:
![Screenshot_1](https://github.com/user-attachments/assets/98c18529-84d3-4212-86fd-8fb3ff25d5a2)
![Screenshot_2](https://github.com/user-attachments/assets/bf1a9d46-4017-4cac-8d0d-707b2df0ddb8)

Let me know if I can help with anything else.

**Reply by limbonaut** [reply]
Cool, thanks!
